### PR TITLE
Use index.php for serving content

### PR DESCRIPTION
**Pull request by chillu**
See https://github.com/silverstripe/silverstripe-framework/issues/7430

I've added a test for operation with SS4. Since `Director::$dev_servers` has been removed, `localhost` is no longer auto-detected as a "dev" environment. This means that accessing `http://localhost/dev` will actually land you on the `Security/login` page. Instead of fixing that, I've simply changed the assertion to check the `Security/login` page contents in the first place. Alternatively, we might want to enforce running Travis in dev environment type through `framework/tests/bootstrap/environment.php`, but that's a pretty broad change I don't want to make right now.

### CODE REVIEW

**Comment by dhensby** [comment]
@tractorcow has recently opened a few PRs forcing travis into dev mode eg: https://github.com/silverstripe/silverstripe-framework/pull/7417

It's probably something we should do because ratelimitting is on in all other environment types for SS 4

File: composer.json [hunk 1]
```diff
@@ -3,9 +3,10 @@
     "description": "Connects the PHP development server to SilverStripe",
     "license": "MIT",
     "require": {
-        "php": "^5.4||^7.0",
+        "php": "^5.6||^7.0",
         "symfony/process": "^3.0",
-        "silverstripe/framework": ">=3.2@dev"
+        "silverstripe/framework": "4.0.x-dev",
```
Comment (tractorcow) [inline composer.json]: For non-recipes we need to set these constraints to allow for stable tags.

File: tests/ServerTest.php [hunk 2]
```diff
@@ -10,17 +10,29 @@ class ServerTest extends \PHPUnit_Framework_TestCase
     public function testStartStop()
     {
         $factory = new ServerFactory(realpath(__DIR__ . '/..'));
+
+        if (file_exists(BASE_PATH . '/vendor/silverstripe/framework/tests/behat/serve-bootstrap.php')) {
+            // SS4
+            $bootstrapFile = 'vendor/silverstripe/framework/tests/behat/serve-bootstrap.php';
+        } else {
+            // SS3
```
Comment (tractorcow) [inline tests/ServerTest.php]: You set the requirements to >4.0, so how could this still support ss3?

**Review by tractorcow — CHANGES_REQUESTED** [review]

File: tests/bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+if (file_exists(BASE_PATH . '/vendor/silverstripe/framework/tests/bootstrap.php')) {
```
Comment (tractorcow) [inline tests/bootstrap.php]: If it's ok I'm going to centralise this check :)

**Review by tractorcow — CHANGES_REQUESTED** [review]

**Review by tractorcow — APPROVED** [review]

**Comment by tractorcow** [comment]
Sorry, I seem to have dropped the ball on this refactor. I'm reverting back tomorrow with a more minimalist change.

File: tests/ServerTest.php (hunk 2 above)
Comment (chillu) [reply tests/ServerTest.php]: As discussed on Slack. we agree there should be separate release lines for 3.x and 4.x. Somebody has already created the 1.x branch for this repo recently, so I've sent a PR to restrict that release line compat to 3.x: https://github.com/silverstripe/silverstripe-serve/pull/17

**Comment by chillu** [comment]
Just installed this as follows, and works fine:

```
git clone -b pulls/4/request-uri https://github.com/open-sausages/silverstripe-installer.git 4.clean
cd 4.clean
composer config repositories.framework vcs https://github.com/open-sausages/silverstripe-framework.git
composer config repositories.assets vcs https://github.com/open-sausages/silverstripe-assets.git
composer config repositories.recipe-core vcs https://github.com/open-sausages/recipe-core.git
composer require "silverstripe/framework:dev-pulls/4/request-uri as 4.0.x-dev" "silverstripe/assets:dev-pulls/1/remove-base-path-autodetect as 1.0.x-dev" "silverstripe/recipe-core:dev-pulls/1/request-uri as 1.0.x-dev"
```

Run installer, then:

```
composer config repositories.serve vcs https://github.com/open-sausages/silverstripe-serve.git
composer require silverstripe/serve:'dev-pulls/index.php-support'
vendor/bin/serve
```

**Comment by tractorcow** [comment]
Fix for my issues at https://github.com/silverstripe/silverstripe-serve/pull/18

**Pull request merged**